crates/backend/src/main.rs

```diff
@@ -38,9 +42,84 @@ async fn main() -> std::io::Result<()> {
                     .header("Cross-Origin-Opener-Policy", "same-origin")
                     .header("Cross-Origin-Embedder-Policy", "require-corp"),
             )
-            .service(fs::Files::new("/", &frontend_folder).index_file("index.html"))
+            .service(serveFrontend(&frontend_folder))
     })
     .bind(format!("127.0.0.1:{}", port))?
     .run()
     .await
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use actix_web::{
+        http,
+        test,
+        App,
+    };
+    use std::{
+        env,
+        fs::File,
+        io::prelude::*,
+    };
+
+    #[actix_rt::test]
+    async fn test_serve_static_index() {
+        let temp_dir = env::temp_dir();
```

Comment:
As you can see, there is a bit of repetition. For each test the same code creates a temp directory.
I had some troubles with factoring this out to a function. We can tackle that later.